src/composites/FileUploader/FileUploader.tsx

```diff
@@ -1,18 +1,106 @@
-import { Box, Button, Callout } from "@radix-ui/themes";
+import { Flex, Button, Callout, Strong } from "@radix-ui/themes";
+import * as AlertDialog from '@radix-ui/react-alert-dialog';
 import { InfoCircledIcon, FileTextIcon } from "@radix-ui/react-icons";
 import { useFileUpload } from "./use-file-uploader";
 import * as schema from "../../data/schema";
 import { useSetAtom } from "jotai";
 import { State } from "../../state";
 import React, { useState } from "react";
+import "./AlertDialog.css"
 
 
 export interface FileUploaderProps { }
 
+interface ValidationErrorIssue {
+  code: string;
+  expected?: any;
+  received?: any;
+  path: (string | number)[];
+  message: string;
+  unionErrors?: { issues: ValidationErrorIssue[]; name: string }[];
+}
+
 export const FileUploader = () => {
   const [_, selectFile] = useFileUpload();
   const setDataset = useSetAtom(State.dataset);
-  const [errorMessage, setErrorMessage] = useState(""); // State to hold error messages
+  const [fileName, setFileName] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
```

Comment:
can we combine `errorMessage` and `errorDetails` into a single object? Otherwise they can go out of sync, we want to enforce that both get set at the same time.

Reply:
updated